platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -130,65 +139,49 @@ export function ViewportGridProvider({ children, service }) {
        */
       case 'SET_DISPLAYSETS_FOR_VIEWPORTS': {
         const { payload } = action;
-        const viewports = state.viewports.slice();
+        const viewports = new Map(state.viewports);
 
-        // Have the initial id set contain all viewports not updated here
-        const viewportIdSet = new Set();
-        viewports.forEach((viewport, index) => {
-          if (!viewport.viewportId) {
-            return;
-          }
-          const isUpdated = payload.find(
-            newViewport => newViewport.viewportIndex === index
-          );
-          if (isUpdated) {
-            return;
-          }
-          viewportIdSet.add(viewport.viewportId);
-        });
+        payload.forEach(updatedViewport => {
+          const { viewportId, displaySetInstanceUIDs } = updatedViewport;
+
+          const previousViewport = viewports.get(viewportId);
 
-        for (const updatedViewport of payload) {
           // Use the newly provide viewportOptions and display set options
           // when provided, and otherwise fall back to the previous ones.
           // That allows for easy updates of just the display set.
-          const { viewportIndex, displaySetInstanceUIDs } = updatedViewport;
-          const previousViewport = viewports[viewportIndex] || {};
-          const viewportOptions = {
-            ...(updatedViewport.viewportOptions ||
-              previousViewport.viewportOptions),
-          };
+          const viewportOptions = merge(
```

Comment:
Does this need a real merge?  It feels like it might cause mixtures that aren't intended, where one version has a value for key A, and the next one has a value for keyB, but doesn't specifically null out keyA, then keyA still applies.
Not saying it doesn't, just asking.

Reply:
I think it is fine